### ISSUE TITLE
Scrapy csv file has uniform empty rows?

### ISSUE THREAD

**Issue by Ibrahim2311** [scrapy/scrapy#2238]
Note: Originally reported on StackOverflow:https://stackoverflow.com/questions/39477662/scrapy-csv-file-has-uniform-empty-rows

here is the spider:

```
import scrapy
from danmurphys.items import DanmurphysItem

class MySpider(scrapy.Spider):
    name = 'danmurphys'
    allowed_domains = ['danmurphys.com.au']
    start_urls = ['https://www.danmurphys.com.au/dm/navigation/navigation_results_gallery.jsp?params=fh_location%3D%2F%2Fcatalog01%2Fen_AU%2Fcategories%3C%7Bcatalog01_2534374302084767_2534374302027742%7D%26fh_view_size%3D120%26fh_sort%3D-sales_value_30_days%26fh_modification%3D&resetnav=false&storeExclusivePage=false']


    def parse(self, response):        
        urls = response.xpath('//h2/a/@href').extract()
        for url in urls:            
            request = scrapy.Request(url , callback=self.parse_page)      
            yield request

    def parse_page(self , response):
        item = DanmurphysItem()
        item['brand'] = response.xpath('//span[@itemprop="brand"]/text()').extract_first().strip()
        item['name'] = response.xpath('//span[@itemprop="name"]/text()').extract_first().strip()
        item['url'] = response.url     
        return item
```
and here is the items :

```
import scrapy
class DanmurphysItem(scrapy.Item):  
    brand = scrapy.Field()
    name = scrapy.Field()
    url = scrapy.Field()
```
when I run the spider with this command :
`scrapy crawl danmurphys -o output.csv`
the output is like this : 
![p2aal](https://cloud.githubusercontent.com/assets/21281149/18508864/b664d196-7a7e-11e6-9b24-f4da044b9cb0.png)

How can I avoid  these uniform empty rows , (by the way when I save to json there are no empty values) .

**Comment by Ibrahim2311** [comment]
I'm working on Windows 10 , python 2.7 interpreter .

**Comment by redapple** [comment]
@Ibrahim2311 , what version of scrpay are you using? (paste the output of `scrapy version -v`)

**Comment by Ibrahim2311** [comment]
@redapple 
C:\Users\Ibrahim>scrapy version -v
Scrapy    : 1.1.0
lxml      : 3.6.0.0
libxml2   : 2.9.3
Twisted   : 16.4.0
Python    : 3.5.2 |Anaconda 4.1.1 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)]
pyOpenSSL : 16.0.0 (OpenSSL 1.0.2h  3 May 2016)
Platform  : Windows-10-10.0.10586-SP0

**Comment by Ibrahim2311** [comment]
it seems I've installed scrapy for conda with interpreter 3.5
and scrapy 1.1.2 for python 2.7 interpreter 
how can I change it to the 2.7 one ?

**Comment by redapple** [comment]
@Ibrahim2311 , I'm sorry but I cannot help you with Conda installs, I not a Conda user myself.

**Comment by Ibrahim2311** [comment]
OK , it's now working just fine with the python 2.7 interpreter (scrapy 1.1.2)
and thanks to you , since I couldn't notice that it was using conda version .

**Comment by Hey-tom** [comment]
What is the fix for this issue on Scrapy 1.4? I face the same issue.

**Comment by chanduparmar** [comment]
Scrapy       : 1.5.0
lxml         : 4.2.1.0
libxml2      : 2.9.5
cssselect    : 1.0.3
parsel       : 1.4.0
w3lib        : 1.19.0
Twisted      : 17.5.0
Python       : 3.6.4 |Anaconda custom (64-bit)| (default, Jan 16 2018, 10:22:32) [MSC v.1900 64 bit (AMD64)]
pyOpenSSL    : 17.5.0 (OpenSSL 1.0.2o  27 Mar 2018)
cryptography : 2.1.4
Platform     : Windows-10-10.0.16299-SP0

I have same issue
